python/ray/tune/experimental/output.py

```diff
@@ -0,0 +1,585 @@
+from typing import List, Dict, Optional, Tuple, Any, TYPE_CHECKING
+
+import collections
+from dataclasses import dataclass
+import datetime
+from enum import Enum
+import numbers
+import numpy as np
+import os
+import pandas as pd
+from tabulate import tabulate
+import textwrap
+import time
+
+from ray._private.dict import unflattened_lookup
+from ray.air._internal.checkpoint_manager import _TrackedCheckpoint
+from ray.tune.callback import Callback
+from ray.tune.logger import pretty_print
+from ray.tune.result import (
+    AUTO_RESULT_KEYS,
+    EPISODE_REWARD_MEAN,
+    MEAN_ACCURACY,
+    MEAN_LOSS,
+    TIME_TOTAL_S,
+    TIMESTEPS_TOTAL,
+    TRAINING_ITERATION,
+)
+from ray.tune.experiment.trial import Trial
+
+if TYPE_CHECKING:
+    from ray.tune.stopper import Stopper
+
+# defines the mapping of the key in result and the key to be printed in table.
+# Note this is ordered!
+DEFAULT_COLUMNS = collections.OrderedDict(
+    {
+        MEAN_ACCURACY: "acc",
+        MEAN_LOSS: "loss",
+        TRAINING_ITERATION: "iter",
+        TIME_TOTAL_S: "total time (s)",
+        TIMESTEPS_TOTAL: "ts",
+        EPISODE_REWARD_MEAN: "reward",
+    }
+)
+
+VALID_SUMMARY_TYPES = {
+    int,
+    float,
+    np.float32,
+    np.float64,
+    np.int32,
+    np.int64,
+    type(None),
+}
+
+
+class AirVerbosity(Enum):
+    SILENT = 0
+    DEFAULT = 1
+    VERBOSE = 2
+
+    def __lt__(self, b):
+        return self.value < b.value
+
+
+try:
+    class_name = get_ipython().__class__.__name__
+    IS_NOTEBOOK = True if "Terminal" not in class_name else False
+except NameError:
+    IS_NOTEBOOK = False
+
+
+def get_air_verbosity() -> Optional[AirVerbosity]:
+    verbosity = os.environ.get("AIR_VERBOSITY", None)
+    if verbosity:
+        return AirVerbosity(int(verbosity)) if verbosity else None
+
+
+def _get_current_time():
+    return datetime.datetime.fromtimestamp(time.time())
+
+
+def _get_time_str(start_time: float, current_time: float) -> Tuple[str, str]:
+    """Get strings representing the current and elapsed time.
+
+    Args:
+        start_time: POSIX timestamp of the start of the tune run
+        current_time: POSIX timestamp giving the current time
+
+    Returns:
+        Current time and elapsed time for the current run
+    """
+    current_time_dt = datetime.datetime.fromtimestamp(current_time)
+    start_time_dt = datetime.datetime.fromtimestamp(start_time)
+    delta: datetime.timedelta = current_time_dt - start_time_dt
+
+    rest = delta.total_seconds()
+    days = rest // (60 * 60 * 24)
+
+    rest -= days * (60 * 60 * 24)
+    hours = rest // (60 * 60)
+
+    rest -= hours * (60 * 60)
+    minutes = rest // 60
+
+    seconds = rest - minutes * 60
+
+    if days > 0:
+        running_for_str = f"{days:.0f} days, "
+    else:
+        running_for_str = ""
+
+    running_for_str += f"{hours:02.0f}:{minutes:02.0f}:{seconds:05.2f}"
```

Comment:
always print hours and minutes??
I would imagine we compute a vector of [days, hours, minutes, seconds], then we find the first non-zero value from left, and construct a string using all the values starting from there?